app/javascript/src/enrollment.js

```diff
@@ -1,25 +1,33 @@
-var initiallyHiddenEnrollmentPanels = document.getElementsByClassName("initially_hidden_enrollment");
-var enrollmentToggleCheckbox = document.getElementById("display_all_enrollments");
+// For when families home page is refreshed when user on it
+document.addEventListener("DOMContentLoaded", function () {
+    const initiallyHiddenEnrollmentPanels = document.getElementsByClassName("initially_hidden_enrollment");
+    const enrollmentToggleCheckbox = document.getElementById("display_all_enrollments");
+    const enrollmentToggleButton = document.getElementById("display_all_enrollments_btn");
 
-function toggleDisplayEnrollments(event) {
-    if (event.target.checked) {
-        for (var i = 0; i < initiallyHiddenEnrollmentPanels.length; i++) {
-            initiallyHiddenEnrollmentPanels[i].classList.remove("hidden");
-        }
-    } else {
-        for (var i = 0; i < initiallyHiddenEnrollmentPanels.length; i++) {
-            initiallyHiddenEnrollmentPanels[i].classList.add("hidden");
+    enrollmentToggleButton.addEventListener('click', toggleDisplayEnrollments);
+    enrollmentToggleCheckbox.addEventListener('click', toggleDisplayEnrollments);
+
+    function toggleDisplayEnrollments(event) {
+        if (event.target.type == "submit") {
+            enrollmentToggleCheckbox.checked = !enrollmentToggleCheckbox.checked;
         }
-    }
-};
 
-// For when family home page loaded through clicking off of the families index page
-if (enrollmentToggleCheckbox != null || enrollmentToggleCheckbox != undefined) {
-    enrollmentToggleCheckbox.addEventListener('click', toggleDisplayEnrollments);
-};
+        if (enrollmentToggleCheckbox.checked) {
+            for (let i = 0; i < initiallyHiddenEnrollmentPanels.length; i++) {
+                initiallyHiddenEnrollmentPanels[i].classList.remove("hidden");
+                enrollmentToggleButton.innerText = l10n("");
```

Comment:
does the empty string need to be localized? or is there a missing string we need here?